src/install/lockfile.zig

```diff
@@ -813,29 +845,32 @@ const Cloner = struct {
             std.mem.set(Tree, trees[1..], Tree{});
         }
 
-        var builder = Tree.Builder{
+        const builder = &Tree.Builder{
+            .allocator = allocator,
             .name_hashes = name_hashes,
             .list = tree_list,
-            .queue = tree_filler_queue,
-            .resolution_lists = resolutions_lists,
             .resolutions = resolutions_buffer,
-            .allocator = allocator,
             .dependencies = this.lockfile.buffers.dependencies.items,
+            .resolution_lists = resolutions_lists,
+            .queue = undefined,
         };
-        var builder_ = &builder;
+        builder.queue = TreeFiller.init(allocator, builder);
+        try builder.queue.ensureUnusedCapacity(root_resolutions.len);
 
         for (root_resolutions) |package_id| {
             if (package_id >= max) continue;
-
-            try builder.list.items(.tree)[0].processSubtree(
-                package_id,
-                builder_,
-            );
+            try builder.queue.add([2]PackageID{ package_id, 0 });
         }
 
         // This goes breadth-first
-        while (builder.queue.readItem()) |pids| {
-            try builder.list.items(.tree)[pids[1]].processSubtree(pids[0], builder_);
+        while (builder.queue.count() > 0) {
+            var queue = builder.queue;
+            builder.queue = TreeFiller.init(allocator, builder);
```

Comment:
This re-allocates the queue each time

We can reuse the memory between calls and improve performance

Reply:
So we need a separate `PriorityQueue` to prevent packages on the next level down from jumping the queue.

I guess we could introduce a `level` integer field to maintain the desired separation :thinking: